core/editor/src/document/document_message_handler.rs

```diff
@@ -187,14 +189,27 @@ impl MessageHandler<DocumentMessage, &InputPreprocessor> for DocumentMessageHand
 				responses.push_back(FrontendMessage::SetActiveDocument { document_index: self.active_document }.into());
 				responses.push_back(RenderDocument.into());
 			}
-			CloseActiveDocument => {
-				responses.push_back(FrontendMessage::PromptCloseConfirmationModal.into());
+			CloseActiveDocumentWithConfirmation => {
+				responses.push_back(FrontendMessage::PromptConfirmationToCloseDocument { document_index: self.active_document }.into());
```

Comment:
Consider renaming `active_document` to `active_document_index` to parallel `document_index` and clarify the difference between a document object and index.